app/models/role.rb

```diff
@@ -46,6 +45,21 @@ def current_holder_biography
     current_holder["details"]["body"]
   end
 
+  def past_holders
+    @past_holders ||= role_holders(current: false)
+      .reverse
+      .map do |rh|
+        rh.dig("links", "person")
+          &.first
+          &.deep_merge(
+            "details" => {
+            "start_year" => Time.zone.parse(rh["details"]["started_on"]).year,
+            "end_year" => Time.zone.parse(rh["details"]["ended_on"]).year,
+            },
+          )
```

Comment:
An alternative to this (perhaps a more Ruby solution):

```ruby
&.tap do |hash|
  hash["details"]["start_year"] = Time.zone.parse(rh["details"]["started_on"]).year
  hash["details"]["end_year"] = Time.zone.parse(rh["details"]["ended_on"]).year
end
```

`tap` takes a block which executes against the object, and then it returns the object.